### ISSUE TITLE
Fix S5944 FP: When namespace or type is same as function name

### ISSUE THREAD

**Issue by PavlinII** [SonarSource/sonar-dotnet#4347]
### Description

Rule raises issue when namespace is the same as function name.

Rule raises issue when return type is the same as function name.

Rule raises issue when declared type is same as function name

### Repro steps

```
Public Function Data() As Data.Container 'S5944	Do not make use of the implicit return value.
    Dim D As Data.Container    'S5944	Do not make use of the implicit return value.
End Function    

Public Function Data() As Data  'S5944	Do not make use of the implicit return value.
    Dim D As Data  'S5944	Do not make use of the implicit return value.
End Function
```

* SonarLint 4.33.1
* Visual Studio version 2019

**Comment by pavel-mikula-sonarsource** [comment]
Also reported on [community](https://community.sonarsource.com/t/false-positive-with-vbnet-s5944/38619)
